### ISSUE TITLE
Local NonNull inference lost inside lambdas?

### ISSUE THREAD

**Issue by entonio** [typetools/checker-framework#1305]
Hi,
The following code triggers a warning:

```java

package pack.dg;

import org.checkerframework.checker.nullness.qual.Nullable;

public class NewClass2 {

    private @Nullable String nullableText = null;

    void trigger() {
        String text = nullableText;
        if (text != null) {
            startStory(text); // no warning

            resumeStory(text,
                    () -> startStory(text) // warning
            );
        }
    }

    public void setNullableText(@Nullable String nullableText) {
        this.nullableText = nullableText;
    }

    private void startStory(String text) {
    }

    private void resumeStory(String text, Runnable alternative) {
    }

}
```

Output:
```
/Users/.../src/main/java/pack/dg/NewClass2.java:16: warning: [argument.type.incompatible] incompatible types in argument.
                    () -> startStory(text) // warning
                                     ^
  found   : @Initialized @Nullable String
  required: @Initialized @NonNull String
1 warning

```

I don't know for sure, but I think this behaviour is unexpected. The local variable just can't change value between the two calls, why is it good for the one outside the lambda but not for the other? I understand there may be an internal CF reason for that, but for the end user it's weird. 

Can you look into it? Thanks!

**Comment by smillst** [comment]
Thanks for reporting this.  You are correct that the warning is a false positive.  Local variable type refinement is ignored within lambda bodies. We are tracking this problem in  #1248, so I'm going to close this issue as a duplicate.
